screens/booking/AllBookings.tsx

```diff
@@ -0,0 +1,142 @@
+import { View, StyleSheet, FlatList } from "react-native";
+
+import CompletedTimeSlot from "../../components/booking/CompletedTimeSlot";
+
+const mock = [
+  {
+    licensePlate: "กข 1334",
+    space: "space 4c",
+    date: "20/01/41",
+    time: "03:38 pm",
+    price: "200",
+  },
+  {
+    licensePlate: "กข 1334",
+    space: "space 4c",
+    date: "20/01/41",
+    time: "03:38 pm",
+    price: "200",
+  },
+  {
+    licensePlate: "กข 1334",
+    space: "space 4c",
+    date: "20/01/41",
+    time: "03:38 pm",
+    price: "200",
+  },
+  {
+    licensePlate: "กข 1334",
+    space: "space 4c",
+    date: "20/01/41",
+    time: "03:38 pm",
+    price: "200",
+  },
+  {
+    licensePlate: "กข 1334",
+    space: "space 4c",
+    date: "20/01/41",
+    time: "03:38 pm",
+    price: "200",
+  },
+  {
+    licensePlate: "กข 1334",
+    space: "space 4c",
+    date: "20/01/41",
+    time: "03:38 pm",
+    price: "200",
+  },
+  {
+    licensePlate: "กข 1334",
+    space: "space 4c",
+    date: "20/01/41",
+    time: "03:38 pm",
+    price: "200",
+  },
+  {
+    licensePlate: "กข 1334",
+    space: "space 4c",
+    date: "20/01/41",
+    time: "03:38 pm",
+    price: "200",
+  },
+  {
+    licensePlate: "กข 1334",
+    space: "space 4c",
+    date: "20/01/41",
+    time: "03:38 pm",
+    price: "200",
+  },
+  {
+    licensePlate: "กข 1334",
+    space: "space 4c",
+    date: "20/01/41",
+    time: "03:38 pm",
+    price: "200",
+  },
+  {
+    licensePlate: "กข 1334",
+    space: "space 4c",
+    date: "20/01/41",
+    time: "03:38 pm",
+    price: "200",
+  },
+  {
+    licensePlate: "กข 1334",
+    space: "space 4c",
+    date: "20/01/41",
+    time: "03:38 pm",
+    price: "200",
+  },
+  {
+    licensePlate: "กข 1334",
+    space: "space 4c",
+    date: "20/01/41",
+    time: "03:38 pm",
+    price: "200",
+  },
+  {
+    licensePlate: "กข 1334",
+    space: "space 4c",
+    date: "20/01/41",
+    time: "03:38 pm",
+    price: "200",
+  },
+  {
+    licensePlate: "กข 1334",
+    space: "space 4c",
+    date: "20/01/41",
+    time: "03:38 pm",
+    price: "200",
+  },
+];
+
+interface allBookingProps { }
+
+const AllBookings: React.FC<allBookingProps> = () => {
+  return (
+    <View style={styles.container}>
+      <FlatList
+        data={mock}
+        renderItem={({ item }) => (
+          <CompletedTimeSlot
+            licensePlate={item.licensePlate}
+            space={item.space}
+            date={item.date}
+            time={item.time}
+            price={item.price}
+          />
+        )}
+        showsVerticalScrollIndicator={false}
+      />
+    </View>
+  );
+};
```

Comment:
From my understanding, you will fetch the data only once and then showing the items that will display. Thus, the response will be very huge doesn't it.

Reply:
Yes, but flatlist will handle the performance issue. Because it will render only the thing that is near the screen. The first batch also render only 10 items.

Also, I haven't impliment the loading logic yet.